src/pagination/pagination.spec.ts

```diff
@@ -267,6 +267,132 @@ describe('ngb-pagination', () => {
            expect(classList).toContain('pagination-123');
          });
        })));
+
+    it('should render and respond to maxSize change correctly', async(inject([TestComponentBuilder], (tcb) => {
+         const html =
+             '<ngb-pagination collectionSize="70" [page]="page" [maxSize]="maxSize" [ellipses]="ellipses"></ngb-pagination>';
+
+         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
+           fixture.detectChanges();
+           expectPages(fixture.nativeElement, ['-« Previous', '+1', '2', '3', '4', '5', '6', '7', '» Next']);
+
+           // disabling ellipsis
+           fixture.componentInstance.ellipses = false;
+
+           // limiting to 3 page numbers
+           fixture.componentInstance.maxSize = 3;
+           fixture.detectChanges();
+           expectPages(fixture.nativeElement, ['-« Previous', '+1', '2', '3', '» Next']);
+
+           fixture.componentInstance.page = 3;
+           fixture.detectChanges();
+           expectPages(fixture.nativeElement, ['« Previous', '1', '2', '+3', '» Next']);
+
+           fixture.componentInstance.page = 4;
+           fixture.detectChanges();
+           expectPages(fixture.nativeElement, ['« Previous', '+4', '5', '6', '» Next']);
+
+           fixture.componentInstance.page = 7;
+           fixture.detectChanges();
+           expectPages(fixture.nativeElement, ['« Previous', '+7', '-» Next']);
```

Comment:
This doesn't look right. Shouldn't we get `['« Previous', '5', '6', '+7', '-» Next']` in this case?